src/highdicom/map/sop.py

```diff
@@ -0,0 +1,789 @@
+                src_img.LossyImageCompressionRatio
+            )
+            self.LossyImageCompressionMethod = (
+                src_img.LossyImageCompressionMethod
+            )
+        self.SamplesPerPixel = 1
+        self.PhotometricInterpretation = 'MONOCHROME2'
+        self.BurnedInAnnotation = 'NO'
+        if contains_recognizable_visual_features:
+            self.RecognizableVisualFeatures = 'YES'
+        else:
+            self.RecognizableVisualFeatures = 'NO'
+        self.ContentLabel = 'ISO_IR 192'  # UTF-8
+        self.ContentDescription = content_description
+        if content_creator_name is not None:
+            check_person_name(content_creator_name)
+        self.ContentCreatorName = content_creator_name
+        self.PresentationLUTShape = 'IDENTITY'
+
+        # Physical dimensions of the image should match those of the source
+
+        # Multi-Frame Functional Groups and Multi-Frame Dimensions
+        shared_fg_item = Dataset()
+        self.SharedFunctionalGroupsSequence = []
+        if pixel_measures is None:
+            if is_multiframe:
+                src_shared_fg = src_img.SharedFunctionalGroupsSequence[0]
+                pixel_measures = src_shared_fg.PixelMeasuresSequence
+            else:
+                pixel_measures = PixelMeasuresSequence(
+                    pixel_spacing=[float(v) for v in src_img.PixelSpacing],
+                    slice_thickness=float(src_img.SliceThickness),
+                    spacing_between_slices=src_img.get(
+                        'SpacingBetweenSlices', None
+                    ),
+                )
+        if is_multiframe:
+            if self._coordinate_system == CoordinateSystemNames.SLIDE:
+                source_plane_orientation = PlaneOrientationSequence(
+                    coordinate_system=self._coordinate_system,
+                    image_orientation=[
+                        float(v) for v in src_img.ImageOrientationSlide
+                    ],
+                )
+            else:
+                src_sfg = src_img.SharedFunctionalGroupsSequence[0]
+                source_plane_orientation = src_sfg.PlaneOrientationSequence
+        else:
+            source_plane_orientation = PlaneOrientationSequence(
+                coordinate_system=self._coordinate_system,
+                image_orientation=[
+                    float(v) for v in src_img.ImageOrientationPatient
+                ],
+            )
+        if plane_orientation is None:
+            plane_orientation = source_plane_orientation
+        self._plane_orientation = plane_orientation
+        self._source_plane_orientation = source_plane_orientation
+
+        # TODO: Double check correctness of DimensionIndexSequence, see
+        # discussion on Segmentation
+        # Also TODO: This includes some segmentation-specific stuff. Fix that
+        self.DimensionIndexSequence = DimensionIndexSequence(
+            coordinate_system=self._coordinate_system
+        )
```

Comment:
Just a comment to confirm that this is indeed wrong as it contains the `SegmentNumber` hardcoded as a dimension, which does not apply in this case

Reply:
This is now outdated